tools/ci-build/changelogger/src/new_entry.rs

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use anyhow::Context;
+use clap::Parser;
+use smithy_rs_tool_common::changelog::{FrontMatter, Markdown, Reference, Target};
+use smithy_rs_tool_common::git::find_git_repository_root;
+use smithy_rs_tool_common::here;
+use std::path::PathBuf;
+
+#[derive(Parser, Debug, Eq, PartialEq)]
+pub struct NewEntryArgs {
+    /// Target audience for the change
+    #[clap(long)]
+    pub applies_to: Option<Vec<Target>>,
+    /// List of git usernames for the authors of the change
+    #[clap(long = "author")]
+    pub authors: Option<Vec<String>>,
+    /// List of relevant issues and PRs
+    #[clap(long = "ref")]
+    pub references: Option<Vec<Reference>>,
+    /// Whether or not the change contains a breaking change (defaults to false)
+    #[clap(long, action)]
+    pub breaking: bool,
+    /// Whether or not the change implements a new feature (defaults to false)
+    #[clap(long, action)]
+    pub new_feature: bool,
+    /// Whether or not the change fixes a bug (defaults to false)
+    #[clap(long, action)]
+    pub bug_fix: bool,
+    /// The changelog entry message
+    #[clap(long)]
+    pub message: Option<String>,
```

Comment:
Please note in the comment that if no message is provided then the user's editor will be opened so that the can author one.

Reply:
Addressed in cf1d1ea